lib/rides/pages/ride_detail_page.dart

```diff
@@ -380,4 +389,69 @@ class _RideDetailPageState extends State<RideDetailPage> {
     await _ride?.cancel();
     setState(() {});
   }
+
+  _showRequestDialog() {
```

Comment:
Ok I tested something that pointed out some bugs: I tried to search for a ride I had already been approved for by the driver.
1) The first and most obvious bug is that this was possible and showed me the result I had already been approved for. I doubt that the algorithm is going to be able to save us from that happening since I doubt it will be given the necessary information
2) I was given the "Request Ride" option, presumably because the ride of the RideDetailPage got the preview status from the SearchSuggestionPage. Should we have a safeguard against that?
3) This should also not have been the case because the drive offered only 1 seat which was already taken. Sure the algorithm is going to bail us out of that one (if it ever comes), but can't we just call isRidePossible to quickly check that?
4) While I was at it, I noticed that we accidentally display the startTime instead of the endTime in TripOverview :D

Reply:
1./2. yes I also noticed that this is possible. I wasn't sure though if this should be possible, at least somehow, in case you want to change how many people you bring on the ride, maybe it's better to add something on the ride-detail-page so you can ask for changes. What do you think?
3. yes we could add isRidePossible there so that drives that are already full can't be requested/are not displayed.
4. should i change that or is that for another PR?

Reply:
1./2. I think it should just not be shown in the list view. What we do with the ride detail is another story, but when it's not in the list then at least it cannot ever be in preview state.
4. Maybe just fix it here, it's related at the least.

Reply:
1./2. okay maybe we should just check if there exists another trip at the same time like for create drive. That would also solve it.

Reply:
and then this is maybe best in another PR since this is the search.